GenericsHomework.Tests/NodeTests.cs

```diff
@@ -0,0 +1,358 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace GenericsHomework.Tests
+{
+
+    [TestClass]
+    public class NodeTests
+    {
+
+        public string ToString(int[] array) => "{" + string.Join(", ", array) + "}";
+        public string ToString(ArrayList list) => "{" + string.Join(", ", list.ToArray()) + "}";
+        public string ToString<T>(List<T> list) => "{" + string.Join(", ", list.ToArray()) + "}";
+
+
+        public Node<int> CreateInt4NodeList(int num1, int num2, int num3, int num4)
+        {
+            Node<int> node1 = new Node<int>(num1);
+            Node<int> node2 = node1.Insert(num2);
+            Node<int> node3 = node2.Insert(num3);
+            Node<int> node4 = node3.Insert(num4);
+            return node1;
+        }
+
+        public Node<string> CreateString4NodeList(string string1, string string2, string string3, string string4)
+        {
+            Node<string> node1 = new Node<string>(string1);
+            Node<string> node2 = node1.Insert(string2);
+            Node<string> node3 = node2.Insert(string3);
+            Node<string> node4 = node3.Insert(string4);
+            return node1;
+        }
+
+        public Node<int[]> CreateIntArray4NodeList(int[] intArr1, int[] intArr2, int[] intArr3, int[] intArr4)
+        {
+            Node<int[]> node1 = new Node<int[]>(intArr1);
+            Node<int[]> node2 = node1.Insert(intArr2);
+            Node<int[]> node3 = node2.Insert(intArr3);
+            Node<int[]> node4 = node3.Insert(intArr4);
+            return node1;
+        }
+
+        [TestMethod]
+        public void NewInstance_GivenInt_NodeContainsInt()
+        {
+            Node<int> node1 = new(789);
+
+            Assert.AreEqual("789", node1.Data.ToString());
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void NewInstance_GivenNullValue_ThrowArgumentNullException()
+        {
+            Node<string> node1 = new(null!);
+        }
+
+        [TestMethod]
+        public void Add_Given4StringNodes_EqualToExpectedString()
+        {
+            Node<string> list = new("First");
+            list.Add("Second");
+            list.Add("Third");
+            list.Add("Fourth");
+            string actual = $"{list.ToString()} {list.Next.ToString()} {list.Next.Next.ToString()} {list.Next.Next.Next.ToString()} {list.Next.Next.Next.Next.ToString()}"; 

```

Comment:
```suggestion
            string actual = $"{list} {list.Next} {list.Next.Next} {list.Next.Next.Next} {list.Next.Next.Next.Next}"; 
```
`ToString()` will be implicitly called when you are doing string interpolation.